jexl-eval/src/lib.rs

```diff
@@ -138,15 +140,19 @@ impl<'a> Evaluator<'a> {
 
     pub fn eval<'b>(&self, input: &'b str) -> Result<'b, Value> {
         let context = value!({});
-        self.eval_in_context(input, &context)
+        // FIXME: we create the parser internally in eval() to minimize changes in function signatures, tests, etc.
+        // For our use case we don't use this function, but maybe makes sense to move this to function parameter
+        let parser = Parser::new();
+        self.eval_in_context(&parser, input, &context)
     }
 
     pub fn eval_in_context<'b, T: serde::Serialize>(
         &self,
+        parser: &Parser,
```

Comment:
We should make the parser be an argument to ::new() and add it as a field on the evaluator. This will be a breaking change, but its a better design.

Reply:
Thank you very much for you feedback! We'll have a look and come back...